### ISSUE TITLE
23666679: Status bar view controller is wrong when a presentation adapts

### ISSUE THREAD

**Issue by openradar-mirror** [lionheart/openradar-mirror#6903]
#### Description

This is a duplicate of rdar://22564923

Summary:
Using the default `UIViewControllerBasedStatusBarAppearance`, the status bar style, visibility and update animation are specified by a view controller. When a view controller is presented, whether is becomes the status bar view controller depends on its presentation style. For example, view controllers presented full screen become the status bar view controller, while those in popovers do not.

UIKit seems to determine the status bar view controller at presentation time, and does not update this when a presentation adapts to a different style.

Steps to Reproduce:
• Open and run the attached sample project on iPhone 6 Plus or iPad Air 2 with Split View enabled (because on these devices the trait collections change, resulting in adaptivity).
• Tap to present a view controller.
• Change size class (rotate).

Some specific steps mentioned below:
1. While in landscape, tap the screen to present
2. Rotate to portrait
3. Tap to dismiss
4. Tap to present
5. Rotate to landscape
6. Tap to dismiss

Expected Results:
• When only the dark green master view is visible, the status bar should have light text.
• When the pink detail view is presented in a popover, the status bar should have light text.
• When the pink detail view is presented full screen, the status bar should have dark text.
• These should remain true after rotation.

Actual Results:
• After step 2, the status bar is light over the full screen pink detail view.
• After step 5, the status bar is dark over the dark green master view (while the pink detail view is in a popover).
• After step 6, the status bar remains dark over the dark green master view. Note that the presented view controller is being kept alive by the presenting view controller somewhat unexpectedly.

Version:
iOS 9, Xcode 7.0 beta 6 (7A192o)

Notes:
It seems that UIKit does not call `_setPresentedStatusBarViewController:` when adapting. As an uninformed outsider, I wonder why UIKit uses an instance variable to store the presented status bar view controller and doesn’t recompute it each time.

Workaround: in the delegate method `presentationController:willPresentWithAdaptiveStyle:transitionCoordinator:`, call `_setPresentedStatusBarViewController:` with what we think the status bar view controller should be.

Configuration:
iPad Air 2 simulator and iPhone 6 Plus simulator

Attachments:
'status-bar-adapt.zip' was successfully uploaded.
## 

Product Version: 9.2 beta 4
Created: 2015-11-25T22:42:09.212920
Originated: 2015-11-25T23:42:00
Open Radar Link: http://www.openradar.me/23666679
